data/gml_registry.xsd

```diff
@@ -0,0 +1,69 @@
+                      <xs:documentation>Used in combination with @elementName. If present, a match for this feature type element is only produced if an XML element whose name equals the value of @elementName and whose value equals the value of @elementValue is found in the GML data. For examle: If namespace/@prefix = 'ex', featureType/@elementName = 'typeOfFeature', and featureType/@elementValue = 'FT', then a match is produced if &lt;ex:typeOfFeature>FT&lt;/ex:typeOfFeature> is present in the GML data.</xs:documentation>
+                    </xs:annotation>
+                  </xs:attribute>
+                  <xs:attribute name="gfsSchemaLocation" type="xs:string">
+                    <xs:annotation>
+                      <xs:documentation>OGR .gfs file in which the feature type is defined (possibly together with other feature types). The GML driver will parse the layer definitions from that file. The schema location can be given as URL (http and https), absolute file path, and relative file path (relative to the location of the registry file).</xs:documentation>
+                    </xs:annotation>
+                  </xs:attribute>
+                  <xs:attribute name="schemaLocation" type="xs:string">
+                    <xs:annotation>
+                      <xs:documentation>XML schema file in which the feature type is defined (possibly together with other feature types). The GML driver will attempt to parse it and create layer definitions. The schema location can be given as URL (http and https), absolute file path, and relative file path (relative to the location of the registry file).</xs:documentation>
+                    </xs:annotation>
+                  </xs:attribute>
+                </xs:complexType>
+              </xs:element>
+            </xs:sequence>
+            <xs:attribute name="prefix" type="xs:string" use="required">
+              <xs:annotation>
+                <xs:documentation>Namespace prefix assigned to the features of this namespace in actual GML data. The prefix will be looked up in the first bytes of the GML file (e.g. xmlns:abc="http://example.org/abc"). NOTE: Case is important - the parser looks for the exact string, i.e., 'ex' is treated differently than 'EX'. If actual data may use different prefixes for the same namespace URI, multiple namespace elements must be added to the gml_registry - one for each applicable prefix.
+
+TBD: Is the prefix actually required? In the gdal/data/gml_registry.xml, the namespace for the japanese schemas does not contain a prefix. Not sure what the exact behavior is when a prefix is defined vs. when it is not. If no prefix is defined, does the parser then automatically resolve namespace prefixes and only match XML elements whose namespace-uri equals the defined uri, and local-name equals featureType/@elementName (plus the test for @elementValue)? Or is the namespace ignored completely, then, and just the local-name matters?</xs:documentation>
```

Comment:
the namespace prefix, when present, is used to confirm the match. If it is missing, then only the uri is used. I don't recall the rationale for that. In any case, no "smart" XML resolution is done. This is just "stupid" string comparisons. The GML driver is not really namespace aware and if you use 2 different namespace prefixes that are for the same namespace URI, it might be confused and doesn't realize that actually define the same namespace